samples/guestbook/gcp/deploy.go

```diff
@@ -0,0 +1,159 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// The deploy program builds the Guestbook server locally and deploys it to
+// GKE.
+
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+func main() {
+	guestbookDir := flag.String("guestbook_dir", "..", "directory containing the guestbook example")
+	flag.Parse()
+	if err := deploy(*guestbookDir); err != nil {
+		fmt.Fprintln(os.Stderr, "deploy:", err)
+		os.Exit(1)
+	}
+}
+
+func deploy(guestbookDir string) error {
+	tfStatePath := filepath.Join(guestbookDir, "gcp", "terraform.tfstate")
+	keys := []string{"project", "cluster_name", "cluster_zone", "bucket", "database_instance", "database_region", "motd_var_config", "motd_var_name"}
+	tfState := make(map[string]string)
+	for _, k := range keys {
+		cmd := exec.Command("terraform", "output", "-state", tfStatePath, k)
```

Comment:
Ah, this is one place where we can make the Go version more efficient: use the [`-json`](https://www.terraform.io/docs/commands/output.html#usage) flag to get this all at once and use a struct to unmarshal the object.

Reply:
Done.